### PR TITLE
feat: sentieon

### PR DESCRIPTION
**Pull request by hannasoderstrom1**
Added functions to get bam input and get vcf input to allow inputs from sentieon.

I can get dry run to work with local output_files.yaml, config.yaml and resources.yaml, using sentieon as aligner.

### CODE REVIEW

File: workflow/Snakefile [hunk 1]
```diff
@@ -427,31 +452,53 @@ use rule mosdepth from qc as qc_mosdepth with:
...
+                    for ext in config.get("picard_collect_multiple_metrics", {}).get("output_ext", [""])
+                ]
+            ),
+            config=lambda wildcards: config["multiqc"]["reports"][wildcards.report]["config"],
+        params:
+            extra=lambda wildcards, input: "--replace-names "
+            + " --sample-names "
+            + " -c "
+            + input.config,
+            use_input_files_only=config.get("multiqc", {}).get("use_input_files_only", True),
```
Comment (nnneeennn) [inline workflow/Snakefile:502]:
```suggestion


```

File: workflow/Snakefile [hunk 2]
```diff
@@ -427,31 +452,53 @@ use rule mosdepth from qc as qc_mosdepth with:
...
+            sample_order="qc/multiqc/sample_order_{report}.tsv",
+        params:
+            extra=lambda wildcards, input: "--replace-names "
+            + input.sample_replacement
+            + " --sample-names "
+            + input.sample_order
+            + " -c "
+            + input.config,
+            use_input_files_only=config.get("multiqc", {}).get("use_input_files_only", True),
+
```
Comment (nnneeennn) [inline workflow/Snakefile:480]:
```suggestion


```

**Comment by nnneeennn** [comment]
you should also add
aligner: "" #bwa_gpu or bwa_sentieon 
 to the config/config.yaml file

**Review by nnneeennn — CHANGES_REQUESTED** [review]

File: workflow/rules/common.smk [hunk 3]
```diff
@@ -55,6 +55,40 @@ def type_generator(types):
...
         return types
 
+def get_bam_input(wildcards, t_n=None, use_sample_wildcard=True):
+    if use_sample_wildcard is True and t_n is None:
+        sample_str = "{}_{}".format(wildcards.sample, wildcards.type)
+    elif use_sample_wildcard is True and t_n:
+        sample_str = "{}_{}".format(wildcards.sample, t_n)
+    else:
+        sample_str = wildcards.file
+
```
Comment (nnneeennn) [inline workflow/rules/common.smk:65]:
```suggestion


```

File: workflow/Snakefile [hunk 4]
```diff
@@ -7,41 +7,55 @@ __license__ = "GPL-3"
...
 ruleorder: annotation_simple_sv_annotation_tn > misc_bgzip
 ruleorder: _results_pindel_vcf > misc_bgzip
 
+aligner = config.get("aligner", None)
+
+if aligner is None:
+    sys.exit("aligner missing from config, valid options: bwa_gpu or bwa_sentieon")
+
+elif aligner == "bwa_gpu":
+    include: "rules/mutectcaller_to_tsv.smk"
```
Comment (nnneeennn) [inline workflow/Snakefile:30]:
```suggestion

    include: "rules/mutectcaller_to_tsv.smk"
```

File: workflow/Snakefile [hunk 5]
```diff
@@ -7,41 +7,55 @@ __license__ = "GPL-3"
...
+    ruleorder: _results_tbi_t > misc_tabix
+    ruleorder: _results_tbi_tn > misc_tabix
+    ruleorder: _results_vcf_aml > misc_bgzip
+    ruleorder: _results_tbi_aml > misc_tabix
+    ruleorder: _results_vcf_all > misc_bgzip
+    ruleorder: _results_tbi_all > misc_tabix
+    ruleorder: parabricks_pbrun_fq2bam > alignment_samtools_index
+    ruleorder: _results_star_crai > alignment_samtools_index
+
+elif aligner == "bwa_sentieon":
```
Comment (nnneeennn) [inline workflow/Snakefile:52]:
```suggestion
elif aligner == "bwa_sentieon":

```

File: workflow/Snakefile [hunk 6]
```diff
@@ -427,30 +455,54 @@ use rule mosdepth from qc as qc_mosdepth with:
...
-        sample_order="qc/multiqc/sample_order_{report}.tsv",
-    params:
-        extra=lambda wildcards, input: "--replace-names "
-        + input.sample_replacement
-        + " --sample-names "
-        + input.sample_order
-        + " -c "
-        + input.config,
-        use_input_files_only=config.get("multiqc", {}).get("use_input_files_only", True),
+if aligner == "bwa_gpu":
```
Comment (nnneeennn) [inline workflow/Snakefile:458]:
```suggestion
if aligner == "bwa_gpu":

```

File: workflow/Snakefile [hunk 7]
```diff
@@ -427,30 +455,54 @@ use rule mosdepth from qc as qc_mosdepth with:
...
+            + input.sample_replacement
+            + " --sample-names "
+            + input.sample_order
+            + " -c "
+            + input.config,
+            use_input_files_only=config.get("multiqc", {}).get("use_input_files_only", True),
+
+
+if aligner == "bwa_sentieon":
+    use rule multiqc from qc as qc_multiqc with:
```
Comment (nnneeennn) [inline workflow/Snakefile:486]:
```suggestion

    use rule multiqc from qc as qc_multiqc with:
```

**Review by nnneeennn — CHANGES_REQUESTED** [review]

File: config/config.yaml [hunk 8]
```diff
@@ -3,6 +3,8 @@ samples: "samples.tsv"
 units: "units.tsv"
 output: "config/output_files.json"
 
+aligner: "bwa_gpu" # bwa_gpu or bwa_sentieon
```
Comment (nnneeennn) [inline config/config.yaml:6]: actually i think we need to add this in .tests/integration/config.yaml file

File: workflow/Snakefile [hunk 9]
```diff
@@ -427,30 +458,56 @@ use rule mosdepth from qc as qc_mosdepth with:
...
-        + " -c "
-        + input.config,
-        use_input_files_only=config.get("multiqc", {}).get("use_input_files_only", True),
+if aligner == "bwa_gpu":
+
+    use rule multiqc from qc as qc_multiqc with:
+        input:
+            files=lambda wildcards: set(
+                [
+                    file.format(sample=sample, type=u.type, lane=u.lane, flowcell=u.flowcell, barcode=u.barcode, read=read, ext=ext)
```
Comment (nnneeennn) [inline workflow/Snakefile:467]:
```suggestion
                    file.format(
                        sample=sample, type=u.type, lane=u.lane, flowcell=u.flowcell, barcode=u.barcode, read=read, ext=ext
                    )
```

File: workflow/rules/common.smk [hunk 10]
```diff
@@ -55,6 +55,43 @@ def type_generator(types):
...
```
Comment (nnneeennn) [inline workflow/rules/common.smk:65]:
```suggestion
```

Comment (nnneeennn) [reply workflow/rules/common.smk:65]: I'm not sure if it has now decided that this should be different or if I marked in the wrong row before

**Review by nnneeennn — CHANGES_REQUESTED** [review]

File: config/config.yaml (hunk 8 above)
Comment (hannasoderstrom1) [reply config/config.yaml:6]: solved in @ 00cb0ea

File: .tests/integration/config.yaml [hunk 11]
```diff
@@ -3,6 +3,8 @@ samples: "samples.tsv"
 units: "units.tsv"
 output: "../../config/output_files.json"
 
+aligner: "bwa_gpu" # bwa_gpu or bwa_sentieon
+
 modules:
   alignment: "906d25d0d4791eb5a54cba9954fddf29e27b85dd"
   annotation: "e58432c3467db31621ab51a16034a026cf2a2c84"
```
Comment (nnneeennn) [inline .tests/integration/config.yaml:10]: sent
```suggestion
  annotation: "e58432c3467db31621ab51a16034a026cf2a2c84"
  sentieon: "develop"
```

**Review by nnneeennn — CHANGES_REQUESTED** [review]

File: workflow/rules/common.smk [hunk 12]
```diff
@@ -55,6 +55,40 @@ def type_generator(types):
     else:
         return types
```
[the remark this answers is not in the document]
Comment (nnneeennn) [reply workflow/rules/common.smk:57]: this still needs to be commited

**Review by nnneeennn — APPROVED** [review]

**Pull request merged**